server/controllers/auth.controller.ts

```diff
@@ -0,0 +1,116 @@
+const bcrypt = require('bcryptjs')
+const { validationResult } = require('express-validator')
+const jwt = require('jsonwebtoken')
+const User = require('../models/User')
+const { SALT_LENGTH, JWT_TOKEN_LIFE_TIME } = require('../config/constants')
+
+class AuthController {
+    static generateAccessToken(id) {
+        const payload = {
+            id,
+        }
+
+        return jwt.sign(payload, process.env.SECRET_KEY, {
+            expiresIn: JWT_TOKEN_LIFE_TIME,
+        })
+    }
+
+    static async hashPassword(password) {
+        return bcrypt.hash(password, SALT_LENGTH)
+    }
+
+    async post(req, res) {
+        try {
+            const { email, password } = req.body
+
+            const errors = validationResult(req)
+            if (!errors.isEmpty()) {
+                return res
+                    .status(400)
+                    .json({ message: 'Uncorrected request', errors })
```

Comment:
you'd better use a middleware for json responses. something like `app.use(express.json());`

and there is one proposal for you. you should send the responses of one structure to a client side. so, for that you can create a function called `formatResponse` which takes three arguments `(data, successMsg, errorMsg)` and transforms them into an object like 
```
data: null | some value,
msg: null | string,
errorMsg: null | string,
```

it will help your FE side handle answers more properly

Reply:
a typo `Uncorrected`, replace it with `Incorrect`

Reply:
fix